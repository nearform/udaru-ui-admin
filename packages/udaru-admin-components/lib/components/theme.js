// Adapted from default theme on https://github.com/jxnblk/styled-system

// breakpoint values
// any array length works with styled-system
const breakpoints = ['500px', '800px', '1200px']

const colors = {
  primary: '#337ab7',
  primaryActive: '#02A171',
  danger: '#d9534f',
  lightGrey: '#D8D8D8',
  lightGrey2: '#888',
  lightGrey3: '#eee',
  darkGrey: '#555',
  black: '#333',
  green: '#5cb85c',
  lightBlue: '#5bc0de',
  orange: '#f0ad4e'
}

export { colors }

export const radius = '4px'

export const maxContainerWidth = '1280px'

// boxShadows
export const boxShadows = [
  `0 0 2px 0 rgba(0,0,0,.08),0 1px 4px 0 rgba(0,0,0,.16)`,
  `0 0 2px 0 rgba(0,0,0,.08),0 2px 8px 0 rgba(0,0,0,.16)`,
  `0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16)`,
  `0 0 2px 0 rgba(0,0,0,.08),0 8px 32px 0 rgba(0,0,0,.16)`
]

export const speeds = {
  slow: '1s',
  medium: '.4s',
  fast: '.2s'
}

export const easings = {
  easeOut: 'cubic-bezier(0,.6,.3,1)',
  expoEaseOut: 'cubic-bezier(0,1,.3,1)',
  bounce: 'cubic-bezier(0,1.2,.5,1.4)'
}

// space is used for margin and padding scales
// it's recommended to use powers of two to ensure alignment
// when used in nested elements
// numbers are converted to px
const space = [0, 4, 8, 16, 32, 64, 128, 256, 512]

// typographic scale
const fontSizes = [12, 14, 16, 20, 24, 32, 48, 64, 96, 128]

// for any scale, either array or objects will work
const lineHeights = [1, 1.125, 1.25, 1.5]

const fontWeights = {
  normal: 500,
  bold: 700
}

const letterSpacings = {
  normal: 'normal',
  caps: '0.25em'
}

// border-radius
const radii = [0, 2, 4, 8]

const borders = [0, '1px solid', '2px solid']

const shadows = [`0 1px 2px 0 ${colors.text}`, `0 1px 4px 0 ${colors.text}`]

const theme = {
  breakpoints,
  colors,
  space,
  fontSizes,
  lineHeights,
  fontWeights,
  letterSpacings,
  radii,
  radius,
  borders,
  shadows,
  boxShadows,
  speeds,
  easings
}

export default theme
