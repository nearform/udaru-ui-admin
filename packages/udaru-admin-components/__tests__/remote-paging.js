import React from 'react'
import renderer from 'react-test-renderer'
import RemotePaging from 'remote-paging'
import { createSerializer } from 'jest-emotion'
import * as emotion from 'emotion'
import theme from '../lib/components/theme'

expect.addSnapshotSerializer(createSerializer(emotion))

jest.mock('react-bootstrap-table', () => {
  return {
    BootstrapTable: props => {
      return (
        <table>
          {' // mocked table output'}
          {props.children}
        </table>
      )
    },
    TableHeaderColumn: props => {
      return (
        <tr>
          <th>{props.dataField + ' // mocked table header column'}</th>
        </tr>
      )
    }
  }
})

it('should render with props', () => {
  const props = {
    data: [],
    udaruUrl: '',
    authorization: '',
    org: '',
    dataTotalSize: 100,
    sizePerPage: 5,
    onPageChange: () => {},
    sizePerPageList: [5, 10, 25, 50, 100],
    currentPage: 1,
    onSizePerPageList: () => {},
    onSelect: () => {},
    searchDelayTime: 300,
    onSearchChange: () => {},
    expandRows: true,
    ExpandComponent: () => <div />,
    expandComponentOnClick: () => {}
  }

  const component = renderer.create(<RemotePaging theme={theme} {...props} />)
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})

it('should always set true to expandable row', () => {
  const component = renderer.create(<RemotePaging theme={theme} />)
  const instance = component.root.instance

  expect(instance.expandableRow()).toBeTruthy()
})

it('should always return ExpandComponent', () => {
  const result = {
    udaruUrl: 'my-udaru-url',
    authorization: 'my-authorization',
    org: 'my-org',
    parentTeamId: 'parent-id',
    parentName: 'parent-name',
    expandComponentOnClick: jest.fn()
  }

  const row = {
    row: {
      id: 1,
      name: 'Team 1'
    }
  }

  const component = renderer.create(
    <RemotePaging
      theme={theme}
      ExpandComponent={() => <div />}
      udaruUrl={result.udaruUrl}
      authorization={result.authorization}
      org={result.org}
      parentTeamId={result.parentTeamId}
      parentName={result.parentName}
      expandComponentOnClick={result.expandComponentOnClick}
    />
  )
  const instance = component.root.instance
  const expandComponentInstance = instance.expandComponent(row)

  const { props } = expandComponentInstance
  const {
    udaruUrl,
    authorization,
    org,
    parentTeamId,
    parentName,
    expandComponentOnClick
  } = props

  expect(udaruUrl).toBe(result.udaruUrl)
  expect(authorization).toBe(result.authorization)
  expect(org).toBe(result.org)
  expect(parentTeamId).toBe(row.row.id)
  expect(parentName).toBe(row.row.name)
  expect(expandComponentOnClick).toBe(result.expandComponentOnClick)
})

it('should open and close row', () => {
  const component = renderer.create(<RemotePaging theme={theme} />)
  const instance = component.root.instance

  const expand = instance.expandColumnComponent({
    isExpandableRow: true,
    isExpanded: true
  })

  const expandedComponent = renderer.create(expand)
  expect(expandedComponent.toJSON()).toMatchSnapshot()

  const collapse = instance.expandColumnComponent({
    isExpandableRow: true,
    isExpanded: false
  })
  const collapedComponent = renderer.create(collapse)
  expect(collapedComponent.toJSON()).toMatchSnapshot()

  const empty = instance.expandColumnComponent({
    isExpandableRow: false
  })
  const emptyComponent = renderer.create(empty)
  expect(emptyComponent.toJSON()).toMatchSnapshot()
})
