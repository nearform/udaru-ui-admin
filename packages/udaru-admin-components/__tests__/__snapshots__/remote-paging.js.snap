// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should open and close row 1`] = `
<div>
  <span
    className="glyphicon glyphicon-remove"
  />
</div>
`;

exports[`should open and close row 2`] = `
<div>
  <span
    className="glyphicon glyphicon-signal"
    style={
      Object {
        "transform": "scale(1, -1) rotate(90deg)",
      }
    }
  />
</div>
`;

exports[`should open and close row 3`] = `
<div>
  ' '
</div>
`;

exports[`should render with props 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <table>
         // mocked table output
        <tr>
          <th>
            id // mocked table header column
          </th>
        </tr>
        <tr>
          <th>
            name // mocked table header column
          </th>
        </tr>
        <tr>
          <th>
            description // mocked table header column
          </th>
        </tr>
        <tr>
          <th>
            usersCount // mocked table header column
          </th>
        </tr>
      </table>
    </div>
  </div>
</div>
`;
